govuk-one-login/authentication-acceptance-tests
# AUT-2546: Remove acceptance test dependency on Stub RPs

## What

This PR removes the deployment pipeline's dependency on our Stub RPs

## Why

The Stub RPs are a convenient tool for testing but they've been coopted into the hot path so they need to be resilient. In reality, we don't care too much about whether the stubs are resilient or not so this dependency conflicts with our desired resilience plan.

## How

This PR creates a local server, using [Spark](https://sparkjava.com/), to act as a callback url for an OIDC flow. The tests now directly route to GOV.UK One Login rather than delegating URL construction and callback to the stub RPs.

This is the same pattern we use in govuk-one-login/authentication-smoke-tests.

---

## How to review

1. Create a new client registry entry in the build environment.
    - It requires only the `openid` scope.
    - Make sure to mark it as a test client and add the appropriate email formats to the test user allowlist.
1. Set RP_CLIENT_ID to the id for the new client
1. Set OP_URL to `https://oidc.build.account.gov.uk/`
1. Set RP_SIGNING_KEY to the base64-encoded form of your private key, removing the header, footer, and newlines.
1. Run tests as normal
